### PR TITLE
Update tests to use pytest better

### PR DESCRIPTION
**Pull request by hmaarrfk**
Needs: https://github.com/vispy/vispy/pull/1774

Which has been folded into this.

### CODE REVIEW

**Comment by djhoese** [comment]
Thanks for looking in to this. It looks like `numpydoc` is still needed by some tests to look at docstrings. Could we use `flake8-docstring` for this?

**Comment by djhoese** [comment]
What do you think about removing the `run_tests_in_object` function which is used when pytest is not imported (lower in the _testing.py module I think)?

**Comment by hmaarrfk** [comment]
> What do you think about removing the run_tests_in_object function which is used when pytest is not imported (lower in the _testing.py module I think)?

depends on how much you want to review

Why is numpydoc simply not a test dependency?

File: appveyor.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ environment:
 
   matrix:
       - PYTHON_VERSION: "2.7"
-        CONDA_DEPENDENCIES: "numpy scipy setuptools nose pytest coverage pytest-cov pytest-sugar pytest-faulthandler cython coveralls mock"
+        CONDA_DEPENDENCIES: "numpy scipy setuptools pytest coverage pytest-cov pytest-sugar pytest-faulthandler cython coveralls mock"
         PIP_DEPENDENCIES: "numpydoc https://files.pythonhosted.org/packages/71/5f/07aad120ca6e4339a5c127631341787bc6bc74add39a1f41223bda949721/freetype_py-2.1.0.post1-py2.py3-none-win_amd64.whl"
       - PYTHON_VERSION: "3.7"
-        CONDA_DEPENDENCIES: "numpy scipy setuptools nose pytest coverage pytest-cov pytest-sugar pytest-faulthandler cython coveralls"
+        CONDA_DEPENDENCIES: "numpy scipy setuptools pytest coverage pytest-cov pytest-sugar pytest-faulthandler cython coveralls"
         PIP_DEPENDENCIES: "numpydoc https://files.pythonhosted.org/packages/71/5f/07aad120ca6e4339a5c127631341787bc6bc74add39a1f41223bda949721/freetype_py-2.1.0.post1-py2.py3-none-win_amd64.whl"
```
Comment (hmaarrfk) [inline appveyor.yml]: Why do you need this numpydoc dependency?

File: .travis.yml [hunk 2]
```diff
@@ -105,7 +105,7 @@ install:
     - conda create -n testenv --yes --quiet pip python=$PYTHON;
     - source activate testenv;
     # add needed packages common to all backends/tests
-    - conda install --yes --quiet numpy scipy nose pytest cython coveralls pytest-cov pytest-sugar pytest-faulthandler flake8;
+    - conda install --yes --quiet numpy scipy pytest cython coveralls pytest-cov pytest-sugar pytest-faulthandler flake8;
```
Comment (hmaarrfk) [inline .travis.yml]:
```suggestion
    - conda install --yes --quiet numpy scipy pytest cython coveralls pytest-cov pytest-sugar pytest-faulthandler flake8 numpydoc;
```

File: appveyor.yml (hunk 1 above)
Comment (hmaarrfk) [reply appveyor.yml]: i see, you needed bleeding edge at one point. Seems like both freetype-py and numpydoc got released on conda-forge as you required.

**Comment by hmaarrfk** [comment]
now that you have embraced the wild wild west of conda-forge, you should be able to fully test everything on conda soon: https://github.com/conda-forge/staged-recipes/pull/10342

**Comment by djhoese** [comment]
> you should be able to fully test everything on conda soon

You mean, no pip installs? At first I thought you meant run vispy tests on the conda-forge builds.

> Why is numpydoc simply not a test dependency?

Just missed dependencies.

> depends on how much you want to review

I assumed it would be a small change along with the other import changes you did. Since pytest is required I thought it made sense and was going to do it myself, but since you're already changing this I thought I'd ask.

Thanks again.

**Comment by djhoese** [comment]
The other thing about numpydoc is that for rendering the sphinx docs we could probably just use napolean that now comes with sphinx.

**Comment by djhoese** [comment]
Apparently flake8-docstring is much more strict that numpydoc. If you install flake8-docstrings and run the normal flake checks:

```
python -m make test flake | wc -l
6315
```

**Comment by djhoese** [comment]
A lot of this was handled similarly in another PR by @larsoner, but there were some things in here I didn't want to lose. The stuff discussed here as alternative solutions (flake8-docstrings, napolean, etc), we can handle in another PR.

**Pull request merged**